### PR TITLE
ci: Cache `re2` dependency

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/actions/setup-node/action.yml [hunk 1]
```diff
@@ -40,8 +40,11 @@ runs:
       with:
         timeout_minutes: 10
         max_attempts: 3
-        command: yarn install --frozen-lockfile --ignore-scripts
+        command: |
+          yarn install --frozen-lockfile --ignore-scripts
+          yarn prepare:install-re2
+          yarn prepare:re2
```
Comment (viceice) [inline .github/actions/setup-node/action.yml:46]: i think we can remove this check and move it to a jest test instead to make things simpler.
Or mode the install call inside the `tools/check-re2.mjs`

File: .github/actions/setup-node/action.yml [hunk 2]
```diff
@@ -40,8 +40,11 @@ runs:
...
 
-    - name: Run scripts
+    - name: Generate files
       shell: bash
-      run: yarn prepare
+      run: test -d lib && yarn prepare:generate || exit 0
```
Comment (viceice) [inline .github/actions/setup-node/action.yml:50]: 🤔 why `exit 0`? which false errors can happen here?

Comment (zharinov) [reply .github/actions/setup-node/action.yml:50]: When we do sparse checkout, there is no `lib` folder, so `test -d lib` has error status, but we don't want it to make the entire action fail.

File: package.json [hunk 3]
```diff
@@ -35,8 +35,7 @@
     "prepare": "run-s prepare:*",
     "prepare:husky": "husky install",
     "prepare:generate": "run-s generate:*",
-    "prepare:install-re2": "npm explore re2 -- npm run install",
-    "prepare:re2": "node tools/check-re2.mjs",
+    "prepare:re2": "npm explore re2 -- npm run install && node tools/check-re2.mjs",
```
Comment (viceice) [inline package.json:38]: this works on windows?

Comment (zharinov) [reply package.json:38]: Just changed cache key, let's see...

Comment (zharinov) [reply package.json:38]: Yes, it worked well

**Pull request merged**